main.go

```diff
@@ -184,103 +124,83 @@ func main() {
 
-	devPortalClient, err := devPortalClientFactory.Create(appleAuthCredentials, cfg.TeamID)
-	if err != nil {
-		failf(err.Error())
-	}
-
-	if err := devPortalClient.Login(); err != nil {
-		failf(err.Error())
-	}
-
-	fmt.Println()
-	logger.TDebugf("Downloading certificates")
-	certs, err := downloadCertificates(certDownloader, logger)
-	if err != nil {
-		failf(err.Error())
-	}
-
-	typeToLocalCerts, err := autocodesign.GetValidLocalCertificates(certs)
-	if err != nil {
-		failf(err.Error())
+	// Auto codesign
+	opts := codesign.Opts{
+		AuthType:                   authType,
+		ShouldConsiderXcodeSigning: false,
+		TeamID:                     cfg.TeamID,
+		ExportMethod:               codesignConfig.DistributionMethod,
+		XcodeMajorVersion:          int(xcodebuildVersion.MajorVersion),
+		RegisterTestDevices:        cfg.RegisterTestDevices,
+		SignUITests:                cfg.SignUITestTargets,
+		MinDaysProfileValidity:     cfg.MinProfileDaysValid,
+		IsVerboseLog:               cfg.VerboseLog,
 	}
 
-	// Create codesign manager
-	manager := autocodesign.NewCodesignAssetManager(devPortalClient, assetWriter, localCodesignAssetManager)
-
-	// Auto codesign
-	distribution := cfg.DistributionType()
 	var testDevices []devportalservice.TestDevice
 	if cfg.RegisterTestDevices && connection != nil {
 		testDevices = connection.TestDevices
 	}
-	codesignAssetsByDistributionType, err := manager.EnsureCodesignAssets(appLayout, autocodesign.CodesignAssetsOpts{
-		DistributionType:        distribution,
-		TypeToLocalCertificates: typeToLocalCerts,
-		BitriseTestDevices:      testDevices,
-		MinProfileValidityDays:  cfg.MinProfileDaysValid,
-		VerboseLog:              cfg.VerboseLog,
-	})
-	if err != nil {
-		if !fallbackProfileDownloader.IsAvailable() {
-			failf(fmt.Sprintf("Automatic code signing failed: %s", err))
-		}
 
-		logger.Println()
-		logger.Warnf("Automatic code signing failed: %s", err)
-		logger.Println()
-		logger.Infof("Falling back to manually managed codesigning assets.")
-		if err := prepareManualAssets(logger, certs, fallbackProfileDownloader, assetWriter); err != nil {
-			failf(fmt.Sprintf("Automatic code signing failed: %s", err))
-		}
-	} else {
-		if err := project.ForceCodesignAssets(distribution, codesignAssetsByDistributionType); err != nil {
-			failf(fmt.Sprintf("Failed to force codesign settings: %s", err))
-		}
+	codesignManager := codesign.NewManagerWithProject(
+		opts,
+		appleAuthCredentials,
+		testDevices,
+		devPortalClientFactory,
+		certDownloader,
+		fallbackProfileDownloader,
+		assetWriter,
+		localCodesignAssetManager,
+		localcodesignasset.NewProvisioningProfileConverter(),
+		project,
+		logger,
+	)
+	_, assets, err := codesignManager.PrepareCodesigning()
+	if err != nil {
+		failf("Failed to prepare codesigning: %s", err)
 	}
 
 	// Export output
 	fmt.Println()
 	logger.Infof("Exporting outputs")
-	if len(codesignAssetsByDistributionType) == 0 {
+	if len(assets) == 0 {
```

Comment:
Now that we added project based filtering for fallback profiles, can this be removed maybe>